src/include/containers/Vector.h

```diff
@@ -248,7 +251,8 @@ bool Vector<ValueType>::operator >=(const Vector& v2) {
 
 template <typename ValueType>
 void Vector<ValueType>::boundaryCheck(int index) const {
-	if (index < 0 || index >= vec.size()) {
+    unsigned long unsignedIndex = index;
+	if (unsignedIndex < 0 || unsignedIndex >= vec.size()) {
```

Comment:
be aware that if you cast a negative signed integer to unsigned, it is simply 'reinterpretation`, i.e. -1 -> 2^32-1 for 32bit int